aws/build.rake

```diff
@@ -255,38 +255,39 @@ def upgrade_frontend(name, host)
   puts IO.read log_path
 end
 
-#
-# The main build task (calls the top-level Rakefile)
-#
-$websites = build_task('websites', [deploy_dir('rebuild'), SHARED_COMMIT_TASK, APPS_COMMIT_TASK]) do
-  Dir.chdir(deploy_dir) do
-    # Lint
-    RakeUtils.system 'rake', 'lint' if CDO.lint
-
-    # Synchronize the Chef cookbooks to the Chef repo for this environment using Berkshelf.
-    if CDO.daemon && CDO.chef_managed
-      Dir.chdir(cookbooks_dir) do
-        old_gemfile = ENV['BUNDLE_GEMFILE']
-        ENV['BUNDLE_GEMFILE'] = File.join(cookbooks_dir, 'Gemfile')
-        begin
-          RakeUtils.bundle_install
-          RakeUtils.bundle_exec 'berks', 'install'
-          RakeUtils.bundle_exec 'berks', 'upload', (rack_env?(:production) ? '' : '--no-freeze')
-          RakeUtils.bundle_exec 'berks', 'apply', rack_env
-        ensure
-          ENV['BUNDLE_GEMFILE'] = old_gemfile
-        end
+# Synchronize the Chef cookbooks to the Chef repo for this environment using Berkshelf.
+task :chef_update do
+  if CDO.daemon && CDO.chef_managed
+    Dir.chdir(cookbooks_dir) do
+      old_gemfile = ENV['BUNDLE_GEMFILE']
+      ENV['BUNDLE_GEMFILE'] = File.join(cookbooks_dir, 'Gemfile')
+      begin
+        RakeUtils.bundle_install
+        RakeUtils.bundle_exec 'berks', 'install'
+        RakeUtils.bundle_exec 'berks', 'upload', (rack_env?(:production) ? '' : '--no-freeze')
+        RakeUtils.bundle_exec 'berks', 'apply', rack_env
+      ensure
+        ENV['BUNDLE_GEMFILE'] = old_gemfile
       end
     end
+  end
+end
 
-    # Build myself
+# Perform a normal local build by calling the top-level Rakefile.
+# Additionally run the lint task if specified for the environment.
+task build: [:chef_update] do
+  Dir.chdir(deploy_dir) do
+    RakeUtils.system 'rake', 'lint' if CDO.lint
     RakeUtils.system 'rake', 'build'
+  end
+end
 
-    # If I'm daemon, do some additional work:
-    if CDO.daemon && rack_env?(:production)
-      # Update the front-end instances, in parallel, but not all at once. When the infrastructure is
-      # properly scaled we should be able to upgrade 20% of the front-ends at a time. Right now we're
-      # over-subscribed (have more resources than we need) so we're restarting 50% of the front-ends.
+# Update the front-end instances, in parallel, but not all at once. When the infrastructure is
```

Comment:
I think this comment is not actually accurate? We are restarting 2 frontends at once, right?